src/GuzzleRequestAdapter.php

```diff
@@ -323,27 +345,31 @@ private function throwFailedResponse(ResponseInterface $response, ?array $errorM
         if ($statusCode >= 200 && $statusCode < 400) {
             return;
         }
-        $statusCodeAsString = (string)$statusCode;
-        if ($errorMappings === null || (!isset($errorMappings[$statusCodeAsString]) &&
+        $statusCodeAsString = "$statusCode";
+        if ($errorMappings === null || (!array_key_exists($statusCodeAsString, $errorMappings) &&
             !($statusCode >= 400 && $statusCode < 500 && isset($errorMappings['4XX'])) &&
             !($statusCode >= 500 && $statusCode < 600 && isset($errorMappings["5XX"])))) {
             $ex = new ApiException("the server returned an unexpected status code and no error class is registered for this code " . $statusCode);
             $ex->setResponseStatusCode($response->getStatusCode());
             throw $ex;
         }
         /** @var array{string,string}|null $errorClass */
-        $errorClass = $errorMappings[$statusCodeAsString] ?? ($errorMappings[$statusCodeAsString[0] . 'XX'] ?? null);
+        $errorClass = array_key_exists($statusCodeAsString, $errorMappings) ? $errorMappings[$statusCodeAsString] : ($errorMappings[$statusCodeAsString[0] . 'XX'] ?? null);
 
         try {
             $rootParseNode = $this->getRootParseNode($response);
-            $error = $rootParseNode->getObjectValue($errorClass);
-            if (is_subclass_of($error, ApiException::class)) {
+            if ($errorClass !== null) {
+                $error = $rootParseNode->getObjectValue($errorClass);
+            } else {
+                $error = null;
+            }
+            if ($error && is_subclass_of($error, ApiException::class)) {
                 $error->setResponseStatusCode($response->getStatusCode());
                 throw $error;
             }
             throw new ApiException("Unsupported error type ". get_debug_type($error));
-        } catch (\RuntimeException $exception){
-            throw new \RuntimeException("", 0, $exception);
+        } catch (RuntimeException $exception){
```

Comment:
Do we need to catch the exception only to throw it again? Maybe no `try-catch` then?